### PR TITLE
Adding zookeeper framework to work with portworx volumes.

### PR DESCRIPTION
**Pull request by piyush-nimbalkar**
Now has support to configure all zk options using env variables.

### CODE REVIEW

File: frameworks/zookeeper/universe/marathon.json.mustache [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    "TASKCFG_ALL_MIN_SESSION_TIMEOUT": "{{zookeeper.min_session_timeout}}",
+    "TASKCFG_ALL_MAX_SESSION_TIMEOUT": "{{zookeeper.max_session_timeout}}",
+    "TASKCFG_ALL_FSYNC_WARNING_THRESHOLD": "{{zookeeper.fsync_warning_threshold}}",
+    "TASKCFG_ALL_SNAP_RETAIN_COUNT": "{{zookeeper.autopurge_snap_retain_count}}",
+    "TASKCFG_ALL_PURGE_INTERVAL": "{{zookeeper.autopurge_interval}}",
+    "TASKCFG_ALL_SYNC_ENABLED": "{{zookeeper.sync_enabled}}",
+    "TASKCFG_ALL_ELECTION_ALGORITHM": "{{zookeeper.election_algorithm}}",
+    "TASKCFG_ALL_INIT_LIMIT": "{{zookeeper.init_limit}}",
+    "TASKCFG_ALL_LEADER_SERVES": "{{zookeeper.leader_serves}}",
+    "TASKCFG_ALL_LEADER_PORT": "{{zookeeper.leader_port}}",
```
Comment (disrani-px) [inline frameworks/zookeeper/universe/marathon.json.mustache]: Should leader port and leader election port be defined in the config too? Also they need to be reserved, so they should be in the port list in svc.yaml

File: frameworks/zookeeper/universe/resource.json [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      "jre-tar-gz": "{{jre-url}}",
+      "libmesos-bundle-tar-gz": "{{libmesos-bundle-url}}",
+      "zookeeper-tar-gz": "http://apache.claz.org/zookeeper/zookeeper-3.4.11/zookeeper-3.4.11.tar.gz",
+      "bootstrap-zip": "{{artifact-dir}}/bootstrap.zip",
+      "scheduler-zip": "{{artifact-dir}}/zookeeper-scheduler.zip",
+      "executor-zip": "{{artifact-dir}}/executor.zip"
+    }
+  },
+  "images": {
+    "icon-small": "https://github.com/dcos/dcos-ui/blob/master/plugins/services/src/img/icon-service-default-small.png?raw=true",
```
Comment (disrani-px) [inline frameworks/zookeeper/universe/resource.json]: Replace with zookeeper logo

File: frameworks/zookeeper/universe/marathon.json.mustache (hunk 1 above)
Comment (piyush-nimbalkar) [reply frameworks/zookeeper/universe/marathon.json.mustache]: it is defined in config, where did you not find it?
I will add it to the port reservation list

File: frameworks/zookeeper/universe/resource.json (hunk 2 above)
Comment (piyush-nimbalkar) [reply frameworks/zookeeper/universe/resource.json]: I could not find any official zookeeper logo except for this - https://zookeeper.apache.org/images/zookeeper_small.gif
and it does not have sizes.
The above size does not adhere to DCOS's image size spec.
If we create our own then we will have to host it somewhere.

File: frameworks/zookeeper/universe/marathon.json.mustache (hunk 1 above)
Comment (disrani-px) [reply frameworks/zookeeper/universe/marathon.json.mustache]: I don't see it in zoo.cfg

File: frameworks/zookeeper/universe/resource.json (hunk 2 above)
Comment (disrani-px) [reply frameworks/zookeeper/universe/resource.json]: We can host it on s3 under the px-dcos bucket. Can be part of a later change.

Comment (piyush-nimbalkar) [reply frameworks/zookeeper/universe/resource.json]: That's a good idea! Will create the images and change it in a separate review when I am done.

File: frameworks/zookeeper/universe/marathon.json.mustache (hunk 1 above)
Comment (piyush-nimbalkar) [reply frameworks/zookeeper/universe/marathon.json.mustache]: The leader port and leader election port are used in the servers list in zoo.cfg, which is created in the ZookeeperConfigUtils.java. So it is indirectly used in the zoo.cfg

Comment (disrani-px) [reply frameworks/zookeeper/universe/marathon.json.mustache]: Got it

**Pull request merged**